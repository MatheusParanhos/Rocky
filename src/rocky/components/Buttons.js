import React, { Component } from "react";
import { Text, View, TouchableOpacity, StyleSheet } from "react-native";
import { deviceWidth, colors, fontFamiliy } from "../../theme/theme";
import FontAwesome from "react-native-vector-icons/FontAwesome";
import Scissors from "./ScissorsButton";
import Paper from "./PaperButton";
import Rock from "./RockButton";
export default class Buttons extends Component {
  render() {
    return (
      <View style={styles.buttonsContainer}>
        
        <View style={styles.bottom}>
          <Rock onPress={this.props.onRockPress}/>
        </View>
        <View style={styles.top}>
          <Paper onPress={this.props.onPaperPress}/>
          <Scissors onPress={this.props.onScissorsPress}/>
        </View>
      </View>
    );
  }
}
const styles = StyleSheet.create({
  buttonsContainer: {
    height: 300,
    width: '100%',
   //  backgroundColor: colors.gray,

   // borderRadius: 25,
   flexDirection:'column'
  },
  top: {
      flex:1,
    height: 100,
    width: deviceWidth / 2,
    flexDirection: "row",
    // backgroundColor: "yellow"
  },
  bottom: {
      flex:1,
    height: 100,
    width: deviceWidth,
   // backgroundColor: "orange",
    alignItems:'center',
    justifyContent:'center',
  }
});
