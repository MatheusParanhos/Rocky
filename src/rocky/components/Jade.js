import React, { Component } from "react";
import { Text, View, TouchableOpacity, StyleSheet, Image } from "react-native";
import { deviceWidth, colors, fontFamiliy } from "../../theme/theme";
import FontAwesome from "react-native-vector-icons/FontAwesome";

export default class Jade extends Component {
    renderIcon = () => {
        if (this.props.jadeChoice === 'scissors') {
            return (
              <FontAwesome
                name={`hand-scissors-o`}
                size={50}
                color={colors.primary}
               // style={{ transform: [{ rotate: "0deg" }] }}
              />
            );
          }
        if (this.props.jadeChoice==='rock' || this.props.jadeChoice==='paper') {
          return (
            <FontAwesome
              name={`hand-${this.props.jadeChoice}-o`}
              size={50}
              color={colors.primary}
              style={{ transform: [{ rotate: "-90deg" }] }}
            />
          );
        }
        
      };
  render() {
    return (
      <View style={styles.container}>
      {this.renderIcon()}
        <Image
          resizeMode={"contain"}
          source={require("../medias/Jade.png")}
          style={styles.imageContainer}
        />
      </View>
    );
  }
}
const styles = StyleSheet.create({
  container: {
    flex: 1,
   // backgroundColor: "red",
   flexDirection:'row',
    alignItems:'center',
    justifyContent:'flex-end'
  },
  imageContainer: {
    width: 150,
    height: 150
  }
});
