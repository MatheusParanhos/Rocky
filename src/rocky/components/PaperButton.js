import React, { Component } from "react";
import { Text, View, TouchableOpacity, StyleSheet } from "react-native";
import { deviceWidth, colors, fontFamiliy } from "../../theme/theme";
import FontAwesome from "react-native-vector-icons/FontAwesome";
import Proptypes from "prop-types";

export default class Paper extends Component {
  render() {
    return (
      <TouchableOpacity
        style={styles.buttonContainer}
        onPress={this.props.onPress}
      >
        <View
          style={{ flex: 1, justifyContent: "center", alignItems: "center" }}
        >
          <FontAwesome name={"hand-paper-o"} size={50} color={colors.primary} />
        </View>
      </TouchableOpacity>
    );
  }
}
Paper.Proptypes = {
  onPress: Proptypes.func.isRequired
};
Paper.defaultProps = {
  onPress: () => {}
}
const styles = StyleSheet.create({
  buttonContainer: {
    height: 50,
    width: deviceWidth / 2,
    justifyContent: "center",
    alignItems: "center",
    borderRadius: 25
  }
});
