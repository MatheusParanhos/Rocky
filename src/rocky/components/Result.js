import React, { Component } from "react";
import { Text, View, TouchableOpacity, StyleSheet } from "react-native";
import { deviceWidth, colors, fontFamily } from "../../theme/theme";
import FontAwesome from "react-native-vector-icons/FontAwesome";
import PropTypes from "prop-types";

export default class Result extends Component {
  renderIcon = () => {
    if (this.props.computerChoice === "scissors") {
      return (
        <FontAwesome
          name={`hand-${this.props.computerChoice}-o`}
          size={50}
          color={colors.primary}
          style={{ transform: [{ rotate: "270deg" }] }}
        />
      );
    }
    if (
      this.props.computerChoice === "rock" ||
      this.props.computerChoice === "paper"
    ) {
      return (
        <FontAwesome
          name={`hand-${this.props.computerChoice}-o`}
          size={50}
          color={colors.primary}
          style={{ transform: [{ rotate: "180deg" }] }}
        />
      );
    }
  };
  renderChoice = () => {
    if (this.props.userChoice === "scissors") {
      return (
        <FontAwesome
          name={`hand-${this.props.userChoice}-o`}
          size={50}
          color={colors.primary}
          style={{ transform: [{ rotate: "90deg" }] }}
        />
      );
    }
    if (this.props.userChoice === "rock" || this.props.userChoice === "paper") {
      return (
        <FontAwesome
          name={`hand-${this.props.userChoice}-o`}
          size={50}
          color={colors.primary}
        />
      );
    }
  };
  render() {
    return (
      <View style={styles.container}>
        <View style={styles.iconContainer}>{this.renderIcon()}</View>
        <Text style={styles.text}>{this.props.countdown}</Text>
        <Text style={styles.text}>{this.props.result}</Text>
        <View
          style={styles.choiceContainer}
        >
          {this.renderChoice()}
        </View>
      </View>
    );
  }
}
Result.propTypes = {
  countdown: PropTypes.number,
  result: PropTypes.string,
  userChoice: PropTypes.string,
  computerChoice: PropTypes.string
};
Result.defaultProps = {
  countdown: 0,
  result: "",
  userChoice: "",
  computerChoice: ""
};

const styles = StyleSheet.create({
  container: {
    height: 200,
    width: deviceWidth,
    justifyContent: "center",
    alignItems: "center",
    borderRadius: 25
  },
  text: {
    fontFamily: fontFamily.splash,
    fontSize: 50
  },
  iconContainer: {
    height: 100,
    width: 100,
    alignItems: "center",
    justifyContent: "center"
  },
  choiceContainer:{
    height: 100,
    width: 100,
    marginTop: 30,
    alignItems: "center",
    justifyContent: "center"
  }
});
