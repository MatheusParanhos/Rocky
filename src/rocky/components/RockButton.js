import React, { Component } from "react";
import { Text, View, TouchableOpacity, StyleSheet } from "react-native";
import { deviceWidth, colors, fontFamiliy } from "../../theme/theme";
import  FontAwesome  from 'react-native-vector-icons/FontAwesome'
import PropTypes from 'prop-types'
export default class Rock extends Component {
  render() {
    return (
      <TouchableOpacity style={styles.buttonContainer} onPress={this.props.onPress}>
        <View
          style={{ flex: 1, justifyContent: "center", alignItems: "center" }}
        >
          <FontAwesome name={'hand-rock-o'} size={50} color={colors.primary}/>
        </View>
      </TouchableOpacity>
    );
  }
}
Rock.PropTypes = {
  onPress: PropTypes.func.isRequired
}
Rock.defaultProps ={
  onPress: () => {}
}
const styles = StyleSheet.create({
  buttonContainer: {
    height: 50,
    width: deviceWidth / 2,
    justifyContent: "center",
    alignItems: "center",
   // backgroundColor: colors.primary,
    borderRadius:25
  }
});
