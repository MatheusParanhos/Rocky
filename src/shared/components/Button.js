import React, { Component } from "react";
import { Text, View, TouchableOpacity, StyleSheet } from "react-native";
import { deviceWidth, colors } from "../../theme/theme";
import PropTypes from 'prop-types'

export class Button extends Component {
  render() {
    return (
      <TouchableOpacity style={styles.buttonContainer} onPress={this.props.onPress}>
        <View
          style={{ flex: 1, justifyContent: "center", alignItems: "center" }}
        >
          <Text style={styles.text}> {this.props.title} </Text>
        </View>
      </TouchableOpacity>
    );
  }
}
Button.propTypes ={
  onPress: PropTypes.func.isRequired
}
Button.defaultProps = {
  onPress: () => {}
}
const styles = StyleSheet.create({
  buttonContainer: {
    height: 50,
    width: deviceWidth / 1.3,
    justifyContent: "center",
    alignItems: "center",
    backgroundColor: colors.primary
  },
  text:{
   // fontFamily: fontFamily.primary,
    color: colors.white,
    fontWeight:'800'
  }
});
