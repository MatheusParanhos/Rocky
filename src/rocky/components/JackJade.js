import React, { Component } from "react";
import { Text, View, TouchableOpacity, StyleSheet, Image } from "react-native";
import { deviceWidth, colors, fontFamiliy } from "../../theme/theme";
import FontAwesome from "react-native-vector-icons/FontAwesome";
import Jack from './Jack'
import Jade from './Jade'
import PropTypes from 'prop-types'

export default class JackJade extends Component {
  render() {
    return (
      <View style={styles.container}>
        <Jack jackChoice={this.props.jackChoice}/>
        <Jade jadeChoice={this.props.jadeChoice}/>
      </View>
    );
  }
}
JackJade.PropTypes ={
  jadeChoice: PropTypes.string,
  jackChoice: PropTypes.string
}
JackJade.defaultProps = {
  jadeChoice: '',
  jackChoice: ''
}

const styles = StyleSheet.create({
  container: {
    flex: .8,
    width:deviceWidth,
    flexDirection:'row'
  },
  imageContainer: {
    width: 150,
    height: 150
  }
});
