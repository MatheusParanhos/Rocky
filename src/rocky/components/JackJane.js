import React, { Component } from "react";
import { Text, View, TouchableOpacity, StyleSheet, Image } from "react-native";
import { deviceWidth, colors, fontFamiliy } from "../../theme/theme";
import FontAwesome from "react-native-vector-icons/FontAwesome";
import Jack from './Jack'
import Jade from './Jade'

export default class JackJane extends Component {
  render() {
    return (
      <View style={styles.container}>
        <Jack jackChoice={this.props.jackChoice}/>
        <Jade jadeChoice={this.props.jadeChoice}/>
      </View>
    );
  }
}
const styles = StyleSheet.create({
  container: {
    flex: .8,
    width:deviceWidth,
  //  backgroundColor: "blue",
    flexDirection:'row'
  },
  imageContainer: {
    width: 150,
    height: 150
  }
});
