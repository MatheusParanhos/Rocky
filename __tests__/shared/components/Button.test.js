import React from 'react';
import {Button} from '../../../src/shared/components/Button';

import renderer from 'react-test-renderer';

test('renders correctly', () => {
  const tree = renderer.create(<Button />).toJSON();
  expect(tree).toMatchSnapshot();
});