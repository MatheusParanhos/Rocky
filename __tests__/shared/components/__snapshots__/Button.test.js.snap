// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "tomato",
      "height": 50,
      "justifyContent": "center",
      "opacity": 1,
      "width": 576.9230769230769,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontWeight": "800",
        }
      }
    >
       
       
    </Text>
  </View>
</View>
`;
